core-client/transports/src/transports/local.rs

```diff
@@ -98,10 +102,11 @@ where
 }
 
 /// Connects to a `Deref<Target = MetaIoHandler<Metadata + Default>`.
-pub fn connect<TClient, THandler, TMetadata>(handler: THandler) -> (TClient, impl Future<Item = (), Error = RpcError>)
+pub fn connect<TClient, THandler, TMetadata, TMiddleware>(handler: THandler) -> (TClient, impl Future<Item = (), Error = RpcError>)
```

Comment:
As @dvdplm noticed it's a breaking change (so would require bumping to `16`), perhaps better to add a new set of methods like `connect_with_middleware` or set the default type somehow?
Usage of middlewares is not super common, so I think it's better to keep the default `connect` stuff simple.